c_emulator/riscv_sim.c

```diff
@@ -1075,3 +1109,56 @@ int main(int argc, char **argv)
   flush_logs();
   close_logs();
 }
+
+int rv_cfg_c_ext_enable(char * isa_str, char * ext_pattern)
```

Comment:
I don't think we really need a full regex engine for this? Currently all this function is being called for is substring matching. Wouldn't it make more sense to parse the arch-string once and tokenize it into a list of of enabled extensions?

Reply:
Hi Alex.  Thanks for the comment.   I was hoping someone would comment on this.

I'm anticipating needing a regex engine for future configuration specifications.  Specifically, 
the RISCV-Config YAML files are more complicated and will need something more than strcmp().
(Think of WARL fields who have ranges that need to be specified.)

So, while the current need could be satisfied with strcmp(),  I decided to get a headstart by
including the pcre2 library in the build and start using it.

Comments?

Reply:
Why can't you use a representation that doesn't need complicated regex parsing? You have the full power of YAML.

Reply:
PCRE indeed is overkill, but given the limited impact (only c_sim) and the possible use in config file parsing, I am inclined to accept this until proven wrong.  We need to discuss in-depth how parsing should work in the long run and may supersede this decision (removing the dependency again).

Reply:
@jrtc27, @arichardson : the schemas for RISC-V configuration options are specified and has been in use for quite a while; I don't propose that we should change them (if that's what you're suggesting).  I'm quite certain that some of the parsing will require a more robust regex parser.

Reply:
By what? Implementations don't use it because they're fixed. Spike and QEMU don't parse it to choose their behaviour. Its whole purpose is to configure the model, but the model doesn't support it, so it surely can't be in real use.

Plus the documentation says, and I quote: "This is still a work in progress and non-backward compatible changes are expected to happen." If the format is crap let's fix the format.

Reply:
Besides, regexes are a terrible choice for parsing ISA strings; good luck writing one that works properly